### DISCUSSION TITLE
[MSVC] Internal compiler error

### DISCUSSION THREAD

**Discussion by skaarj1989** [jrouwe/JoltPhysics#1065] [Q&A] [answered]
I've just updated **VisualStudio 2022** and found this nasty surprise.
`1>E:\source\JoltPhysics\Jolt\Renderer\DebugRenderer.cpp(204): fatal error C1001: Internal compiler error.`

Hopefully, someone on the VisualStudio team fixes this:
https://developercommunity.visualstudio.com/t/Jolt-Physics-C1001:-Internal-compiler-er/10642751

This is a caution for those about to update **VisualStudio 2022**.

**Comment by jrouwe** [comment]
I'm on the latest official released version and it works fine here. Occasionally I get an internal compiler error but usually it goes away if you compile again or do a rebuild all.

**Reply by skaarj1989** [reply]
Unfortunately, it's not that simple, and rebuilding doesn't solve this (that's why I created the issue).

**Reply by jrouwe** [reply]
You may want to add some more detail to your issue on the MSVC site, there are many different configurations that you could compile (there are 10 batch files that produce project files, but you can produce many more build configs) and it would probably help to point them to a particular commit hash too.

If this issue still persists by the time the stable build comes out, I'll try to create a workaround. I don't support beta versions of MSVC.

**Comment by TIHan** [comment]
What version of VS2022 are you using? I can compile fine and I'm on `17.8.3`.

**Reply by skaarj1989** [reply]
17.10.0 Preview 4.0

**Reply by jrouwe** [reply]
17.9.6 works fine too, I think it's just the beta that has a problem.

**Comment by skaarj1989** [comment, marked as answer]
Looks like it's been fixed with the new release `17.10.0 Preview 5.0`
